bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/scheduling/Scheduler.java

```diff
@@ -456,7 +457,8 @@ public FutureValue createFuture(Strand parent, Callback callback, Map<String, Ob
 
     private FutureValue createFuture(Strand parent, Callback callback, Type constraint, Strand newStrand) {
         if (parent != null) {
-            newStrand.observerContext = parent.observerContext;
+            //TODO: improve by using a copy on write map
```

Comment:
Shall we also create an issue for the Todo and link here or fix the Todo before the PR is merged?

Reply:
+1 will create

Reply:
#26710